src/mintpy/los2slope.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env python3
+#########################################################################
+# Program is part of MintPy                                             #
+# Copyright (c) 2013, Zhang Yunjun, Heresh Fattahi                      #
+# Author: Author: Ashok Dahal, Alexander Handwerger, Eric Fielding.     #
+# Mar 2024 Based on previous code by Antonio Valentino, Zhang Yunjun.   #
+#########################################################################
+
+import numpy as np
+from osgeo import gdal
+from scipy.ndimage import gaussian_filter
+
+from mintpy.utils import ptime, readfile, writefile
+
+################################################################################
+
+
+def median_filter(dem, kernel_size):
+    """
+    Apply median filtering to a given Digital Elevation Model (DEM).
+
+    Parameters:
+    - dem: 2D NumPy array representing the DEM
+    - kernel_size: Size of the median filter kernel (odd integer)
+
+    Returns:
+    - filtered_dem: 2D NumPy array representing the filtered DEM
+    """
+    # Ensure the kernel size is odd
+    if kernel_size % 2 == 0:
+        kernel_size += 1
+
+    # Apply median filtering
+    filtered_dem = np.zeros_like(dem)
+    padded_dem = np.pad(dem, kernel_size // 2, mode="reflect")  # Reflect padding
+    for i in range(filtered_dem.shape[0]):
+        for j in range(filtered_dem.shape[1]):
+            window = padded_dem[i : i + kernel_size, j : j + kernel_size]
+            filtered_dem[i, j] = np.median(window)
+
+    return filtered_dem
+
+
+def smooth_dem(dem, sigma):
```

Comment:
I do not see the need to wrap smoothing in the function, as it is directly called from the module, so suggest to remove this function smooth_dem, and replace LINE 308, with 
```
# Apply Gaussian smoothing
 smoothed_dem = gaussian_filter(dem, sigma=sigma)
```